### PR TITLE
Change resources default from inline to cdn

### PR DESCRIPTION
**Pull request by Maggie-M**
issues: closes #2754

### CODE REVIEW

**Comment by Maggie-M** [comment]
@damianavila I ran into an issue with testing this using BOKEH_VERSION: 

```
$ BOKEH_VERSION=0.9.2 python color_scatter.py 
Traceback (most recent call last):
  File "color_scatter.py", line 4, in <module>
    from bokeh.plotting import figure, show, output_file
  File "/Users/mmari/Desktop/continuum/bokeh/bokeh/plotting.py", line 6, in <module>
    from . import _glyph_functions as gf
  File "/Users/mmari/Desktop/continuum/bokeh/bokeh/_glyph_functions.py", line 3, in <module>
    from .models import glyphs, markers, BoxSelectTool
  File "/Users/mmari/Desktop/continuum/bokeh/bokeh/models/__init__.py", line 11, in <module>
    from .map_plots import *
  File "/Users/mmari/Desktop/continuum/bokeh/bokeh/models/map_plots.py", line 13, in <module>
    from .plots import Plot
  File "/Users/mmari/Desktop/continuum/bokeh/bokeh/models/plots.py", line 23, in <module>
    from .widget import Widget
  File "/Users/mmari/Desktop/continuum/bokeh/bokeh/models/widget.py", line 11, in <module>
    from ..embed import notebook_div
  File "/Users/mmari/Desktop/continuum/bokeh/bokeh/embed.py", line 20, in <module>
    from .resources import Resources
  File "/Users/mmari/Desktop/continuum/bokeh/bokeh/resources.py", line 258, in <module>
    INLINE = Resources(mode="inline")
  File "/Users/mmari/Desktop/continuum/bokeh/bokeh/resources.py", line 166, in __init__
    raise ValueError("setting 'version' makes sense only when 'mode' is set to 'cdn'")
ValueError: setting 'version' makes sense only when 'mode' is set to 'cdn'
```

I had to comment out the check in resources.py to be able to test.

Is there anything we should do about this?

**Comment by damianavila** [comment]
Maybe `BOKEH_RESOURCES=cdn BOKEH_VERSION=0.9.2 python color_scatter.py`? Btw, you have a test to fix according to TravisCI...

**Comment by bryevdv** [comment]
@damianavila No, the problem is that `resouces.py` creates `INLINE` as a module level variable as a convenience, and that (alone) is what triggers the exception.

**Comment by bryevdv** [comment]
More info:
```
bryan@0199-bryanv (git:tornado) ~/work/bokeh/bokehjs/src/coffee $ BOKEH_RESOURCES=cdn BOKEH_VERSION=0.9.2 ipython
Python 3.4.3 |Anaconda 2.3.0 (x86_64)| (default, Mar  6 2015, 12:07:41) 
Type "copyright", "credits" or "license" for more information.

IPython 3.2.0 -- An enhanced Interactive Python.
Anaconda is brought to you by Continuum Analytics.
Please check out: http://continuum.io/thanks and https://anaconda.org
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: from bokeh.resources import INLINE
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-1-8dce39b5d7ec> in <module>()
----> 1 from bokeh.resources import INLINE

/Users/bryan/work/bokeh/bokeh/resources.py in <module>()
    256 CDN = Resources(mode="cdn")
    257 
--> 258 INLINE = Resources(mode="inline")

/Users/bryan/work/bokeh/bokeh/resources.py in __init__(self, mode, version, root_dir, minified, log_level, root_url)
    164 
    165         if self.version and not mode.startswith('cdn'):
--> 166             raise ValueError("setting 'version' makes sense only when 'mode' is set to 'cdn'")
    167 
    168         if root_url and not mode.startswith('server'):

ValueError: setting 'version' makes sense only when 'mode' is set to 'cdn'
```

**Comment by damianavila** [comment]
oh... I did not test it... just reading the first post... 
Not sure about what to do here... the `ValueError` is valuable, I think the problem in the definition of `INLINE` itself at the module level, but I also understand the convenience...

**Comment by bryevdv** [comment]
@damianavila yes that is the problem and @mattpap has pointed it out before. There are potentially things we could do to effectively make `CDN` and `INLINE` on demand:

http://stackoverflow.com/questions/2447353/getattr-on-a-module

TLDR: `sys.modules[__name__] = Foo()` is semi-sanctioned bt GvR, so...

**Comment by bryevdv** [comment]
@bokeh/core Is this change for 0.9.3 or 0.10?

**Comment by mattpap** [comment]
> Is this change for 0.9.3 or 0.10?

0.10 for sure.

**Comment by bryevdv** [comment]
@Maggie-M can you re-open against `dev_0.10`

**Comment by damianavila** [comment]
>TLDR: sys.modules[__name__] = Foo() is semi-sanctioned bt GvR, so...

well... if it is semi-sanctioned... I guess we should open another issue for this...

>0.10 for sure.

I agree...
